bootloader/uboot.go

```diff
@@ -60,18 +89,51 @@ func (u *uboot) dir() string {
 	if u.rootdir == "" {
 		panic("internal error: unset rootdir")
 	}
-	if u.noslashboot {
-		return u.rootdir
-	}
-	return filepath.Join(u.rootdir, "/boot/uboot")
+	return filepath.Join(u.rootdir, u.basedir)
 }
 
-func (u *uboot) InstallBootConfig(gadgetDir string, opts *Options) (bool, error) {
+func (u *uboot) InstallBootConfig(gadgetDir string, blOpts *Options) (bool, error) {
 	gadgetFile := filepath.Join(gadgetDir, u.Name()+".conf")
-	systemFile := u.ConfigFile()
-	if opts != nil && opts.Recovery {
-		systemFile = filepath.Join(u.rootdir, "uboot.env")
+	// if the gadget file is empty, then we don't install anything
+	// this is because there are some gadgets, namely the 20 pi gadget right
+	// now, that don't use a uboot.env to boot and instead use a boot.scr, and
+	// installing a uboot.env file of any form in the root directory will break
+	// the boot.scr, so for these setups we just don't install anything
+	// TODO:UC20: how can we do this better? maybe parse the file to get the
+	//            actual format?
+	st, err := os.Stat(gadgetFile)
+	if err != nil {
+		return false, err
+	}
+	if st.Size() == 0 {
+		// we have a uboot.conf, and hence a uboot bootloader in the gadget, so
```

Comment:
// we have an empty uboot.conf,  and hence a uboot ... in the gadget, but nothing to copy in this case and instead ...